### PR TITLE
Blast2.9.0 update

### PR DESCRIPTION
**Pull request by christiam**
Bioconda requires reviews prior to merging pull-requests (PRs). To facilitate this, once your PR is passing tests and ready to be merged, please add the `please review & merge` label so other members of the bioconda community can have a look at your PR and either make suggestions or merge it. Note that if you are not already a member of the bioconda project (meaning that you can't add this label), please ping `@bioconda/core` so that your PR can be reviewed and merged (please note if you'd like to be added to the bioconda project). Please see https://github.com/bioconda/bioconda-recipes/issues/15332 for more details.

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by christiam** [comment]
@bioconda/core - when you have a moment, could you please review this PR? 

It updates the dependencies (removes outdated ones, adds a missing ones) and test script.

File: recipes/blast/build.sh [hunk 1]
```diff
@@ -42,11 +39,8 @@ LIB_INSTALL_DIR=$PREFIX/lib/ncbi-blast+
 # -vdb: disable VDB/SRA toolkit
 # z: set zlib
 # bz2: set libbz2
-# boost: set boost
```
Comment (epruesse) [inline recipes/blast/build.sh]: I take it blast no longer uses boost?

File: recipes/blast/build.sh [hunk 2]
```diff
@@ -42,11 +39,8 @@ LIB_INSTALL_DIR=$PREFIX/lib/ncbi-blast+
...
 # -openssl: disable openssl
 # -gcrypt: disable gcrypt (needed on OSX)
-# gnutls: set gnutls (preferred over openssl for -remote support)
-# nettle: set nettle
```
Comment (epruesse) [inline recipes/blast/build.sh]: Do we still have support for `-remote` then?

File: recipes/blast/build.sh [hunk 3]
```diff
@@ -55,6 +49,7 @@ export AR="${AR} rcs"
 ./configure \
     --with-dll \
     --with-mt \
+    --with-openmp \
```
Comment (epruesse) [inline recipes/blast/build.sh]: Nice. Does that make an impact to performance?

File: recipes/blast/build.sh [hunk 4]
```diff
@@ -68,31 +63,32 @@ export AR="${AR} rcs"
...
     --without-openssl \
-    --with-gnutls=$PREFIX
+    --without-gnutls \
+    --without-gcrypt
 
-projects="algo/blast/ app/ objmgr/ objtools/align_format/ objtools/blast/"
+apps="blastp.exe blastn.exe blastx.exe tblastn.exe tblastx.exe psiblast.exe rpsblast.exe rpstblastn.exe"
+apps="$apps makembindex.exe segmasker.exe dustmasker.exe windowmasker.exe deltablast.exe"
+apps="$apps makeblastdb.exe blastdbcmd.exe blastdb_aliastool.exe convert2blastmask.exe blastdbcheck.exe"
+apps="$apps makeprofiledb.exe blast_formatter.exe"
```
Comment (epruesse) [inline recipes/blast/build.sh]: Long list. Sounds ok to me. What is the benefit of giving apps vs giving projects? I assume you wanted to add something?

Comment (epruesse) [reply recipes/blast/build.sh]: Rewrap to 77 chars per line? Shouldn't make much of a difference, but easier to read here or via email.

File: recipes/blast/build.sh [hunk 5]
```diff
@@ -68,31 +63,32 @@ export AR="${AR} rcs"
...
 
 cd build
-make -j${CPU_COUNT} -f Makefile.flat all_projects="$projects"
+make -j${CPU_COUNT} -f Makefile.flat $apps
 
 # remove temporary link
 rm $LIB_INSTALL_DIR
 
 mkdir -p $PREFIX/bin $LIB_INSTALL_DIR
-rm $SRC_DIR/c++/ReleaseMT/bin/*_unit_test
```
Comment (epruesse) [inline recipes/blast/build.sh]: I take it it no longer installs those? That's good.

File: recipes/blast/build.sh [hunk 6]
```diff
@@ -68,31 +63,32 @@ export AR="${AR} rcs"
...
 cp $SRC_DIR/c++/ReleaseMT/bin/* $PREFIX/bin/
 cp $SRC_DIR/c++/ReleaseMT/lib/* $LIB_INSTALL_DIR
 
 chmod +x $PREFIX/bin/*
 sed -i.bak '1 s|^.*$|#!/usr/bin/env perl|g' $PREFIX/bin/update_blastdb.pl
+sed -i.bak 's/mktemp.*/mktemp`/; /export PATH/ s,$,:\$PATH,; s/exit 1/exit 0/' $PREFIX/bin/get_species_taxids.sh
```
Comment (epruesse) [inline recipes/blast/build.sh]: Put a comment explaining patch? Or use patch?

File: recipes/blast/meta.yaml [hunk 7]
```diff
@@ -10,43 +10,57 @@ source:
...
   host:
-    - boost
     - zlib
     - bzip2
-    - gnutls
-    - gmp
-    - nettle 3.4
     - pcre
+    - entrez-direct
+    - curl
```
Comment (epruesse) [inline recipes/blast/meta.yaml]: So blast needs curl now? Does it link against it? If so, might need to pin.

File: recipes/blast/meta.yaml [hunk 8]
```diff
@@ -10,43 +10,57 @@ source:
...
   run:
-    - boost
     - zlib
     - bzip2
-    - gnutls
-    - gmp
-    - nettle 3.4
     - pcre
```
Comment (epruesse) [inline recipes/blast/meta.yaml]: thinking about it - `pcre` might need to be pinned.

File: recipes/blast/meta.yaml [hunk 9]
```diff
@@ -10,43 +10,57 @@ source:
...
     - {{ compiler('cxx') }}
   host:
-    - boost
     - zlib
     - bzip2
-    - gnutls
-    - gmp
-    - nettle 3.4
     - pcre
+    - entrez-direct
```
Comment (epruesse) [inline recipes/blast/meta.yaml]: What does this one include, actually?

File: recipes/blast/meta.yaml [hunk 10]
```diff
@@ -10,43 +10,57 @@ source:
...
+    - makembindex -version > /dev/null
+    - psiblast -version > /dev/null
+    - rpsblast -version > /dev/null
+    - rpstblastn -version > /dev/null
+    - segmasker -version > /dev/null
+    - tblastn -version > /dev/null
+    - tblastx -version > /dev/null
+    - windowmasker -version > /dev/null
+    - update_blastdb.pl --version > /dev/null
+    - get_species_taxids.sh -h > /dev/null
```
Comment (epruesse) [inline recipes/blast/meta.yaml]: Great - that helps making sure they are all there.

File: recipes/blast/build.sh (hunk 1 above)
Comment (christiam) [reply recipes/blast/build.sh]: No, just the unit tests do (which were being deleted at the end of the recipe)

File: recipes/blast/build.sh (hunk 2 above)
Comment (christiam) [reply recipes/blast/build.sh]: Yes.

File: recipes/blast/build.sh (hunk 3 above)
Comment (christiam) [reply recipes/blast/build.sh]: In some cases, yes.

File: recipes/blast/build.sh (hunk 4 above)
Comment (christiam) [reply recipes/blast/build.sh]: > What is the benefit of giving apps vs giving projects?
This is more specific, it builds only those applications.

> Rewrap to 77 chars per line? 
Will do.

File: recipes/blast/build.sh (hunk 5 above)
Comment (christiam) [reply recipes/blast/build.sh]: That is indeed the case.

File: recipes/blast/build.sh (hunk 6 above)
Comment (christiam) [reply recipes/blast/build.sh]: I'll add a comment shortly.

File: recipes/blast/meta.yaml (hunk 7 above)
Comment (christiam) [reply recipes/blast/meta.yaml]: `update_blastdb.pl` uses it.

File: recipes/blast/meta.yaml (hunk 9 above)
Comment (christiam) [reply recipes/blast/meta.yaml]: This is needed for `get_species_taxids.sh`

**Comment by christiam** [comment]
> Would you like to add yourself to `recipe-maintainers`

Sure, how do I go about doing that?

File: recipes/blast/build.sh (hunk 4 above)
Comment (epruesse) [reply recipes/blast/build.sh]: thanks

File: recipes/blast/meta.yaml (hunk 7 above)
Comment (epruesse) [reply recipes/blast/meta.yaml]: Ah, so cmdline. No pinning then.

**Comment by epruesse** [comment]
> > Would you like to add yourself to `recipe-maintainers`
> 
> Sure, how do I go about doing that?

Add to maintainers. I think we need documentation for that somewhere.

**Comment by christiam** [comment]
@bioconda/core - when you have a moment, could you please review this PR? I cannot add the `please review & merge` label. 

This PR updates the dependencies (removes outdated ones, adds a missing ones) and test script.

**Comment by dpryan79** [comment]
Thanks @christiam!

**Pull request merged**